File: demo/viewjava/src/main/java/pers/shawxingkwok/kdatastore/demo/viewjava/MainActivity.java
package pers.shawxingkwok.kdatastore.demo.viewjava;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatDelegate;
import pers.shawxingkwok.kdatastore.demo.settings.Settings;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.container, MainFragment.newInstance())
                .commitNow();
        }

        // link `isDarkMode` with `theme` via `liveData`
        Settings.isDarkMode().getLiveData().observe(this, (isDark)->{
            int mode;

            if (isDark == null)
                mode = AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
            else if (isDark)
                mode = AppCompatDelegate.MODE_NIGHT_YES;
            else
                mode = (AppCompatDelegate.MODE_NIGHT_NO);

            AppCompatDelegate.setDefaultNightMode(mode);
            getDelegate().applyDayNight();
        });
    }
}

File: demo/viewkt/src/main/java/pers/shawxingkwok/kdatastore/demo/viewkt/MainActivity.kt
package pers.shawxingkwok.kdatastore.demo.viewkt

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.fragment.app.commitNow
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach
import pers.shawxingkwok.kdatastore.demo.settings.Settings

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null)
            supportFragmentManager.commitNow {
                replace(R.id.container, MainFragment.newInstance())
            }

        // link `isDarkMode` with `theme`.
        Settings.isDarkMode.onEach {
            val mode = when(it){
                null -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
                true -> AppCompatDelegate.MODE_NIGHT_YES
                false -> AppCompatDelegate.MODE_NIGHT_NO
            }
            AppCompatDelegate.setDefaultNightMode(mode)
            delegate.applyDayNight()
        }
        .launchIn(lifecycleScope)

        /* or use liveData
        Settings.isDarkMode.liveData.observe(this){
            val mode = when(it){
                null -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
                true -> AppCompatDelegate.MODE_NIGHT_YES
                false -> AppCompatDelegate.MODE_NIGHT_NO
            }
            AppCompatDelegate.setDefaultNightMode(mode)
            delegate.applyDayNight()
        }
        */
    }
}